guide/src/conversions.md

```diff
@@ -1,8 +1,103 @@
+
+| Python        | Rust                            | Rust (Python-native) |
+| ------------- |:-------------------------------:|:--------------------:|
+| `Any`         | -                               | `&PyAny`             |
+| `str`         | `String`, `&str`, `Cow<str>`    | `&PyUnicode`         |
+| `bytes`       | `&[u8]`                         | `&PyBytes`           |
+| `bool`        | `bool`                          | `&PyBool`            |
+| `int`         | Any integer type (`i32`, `u32`, `usize`, etc) | `&PyLong` |
+| `float`       | `f32`, `f64`                    | `&PyFloat`           |
+| `complex`     | `num_complex::Complex`[^1]      | `&PyComplex`         |
+| `Optional[T]` | `Option<T>`                     | -                    |
+| `List[T]`     | `Vec<T>`                        | `&PyList`            |
+| `Dict[K, V]`  | `HashMap<K, V>`, `BTreeMap<K, V>` | `&PyDict`          |
+| `Tuple[T, U]` | `(T, U)`                        | `&PyTuple`           |
+| `Set[T]`      | `HashSet<T>`, `BTreeSet<T>`     | `&PySet`             |
+| `FrozenSet[T]` | `HashSet<T>`, `BTreeSet<T>`    | `&PyFrozenSet`       |
+| `FrozenSet[T]` | `HashSet<T>`, `BTreeSet<T>`    | `&PyFrozenSet`       |
+| `Sequence[T]` | `Vec<T>`                        | `&PySequence`        |
+| `Iterator[Any]` | -                             | `&PyIterator`        |
+| `bytearray`   | -                               | `&PyByteArray`       |
+| `slice`       | -                               | `&PySlice`           |
+| `type`        | -                               | `&PyType`            |
+| `module`      | -                               | `&PyModule`          |
+| `datetime.datetime` | -                         | `&PyDateTime`        |
+| `datetime.date` | -                             | `&PyDate`            |
+| `datetime.time` | -                             | `&PyTime`            |
+| `datetime.tzinfo` | -                           | `&PyTzInfo`          |
+| `datetime.timedelta` | -                        | `&PyDelta`           |
+
+There are also a few special types related to the GIL and Rust-defined `#[pyclass]`es which may come in useful:
+
+| What          | Description |
+| ------------- | ------------------------------- |
+| `Python`      | A GIL token, used to pass to PyO3 constructors to prove ownership of the GIL |
+| `PyObject`    | A Python object isolated from the GIL lifetime. This can be sent to other threads. To call Python APIs using this object, it must be used with `AsPyRef::as_ref` to get a `&PyAny` reference. |
+| `Py<T>`       | Same as above, for a specific Python type or `#[pyclass]` T. |
+| `&PyCell<T>`  | A `#[pyclass]` value owned by Python. |
+| `PyRef<T>`    | A `#[pyclass]` borrowed immutably. |
+| `PyRefMut<T>` | A `#[pyclass]` borrowed mutably. |
+
+For more detail on accepting `#[pyclass]` values as function arguments, see [the section of this guide on Python Classes](class.md).
```

Comment:
Quickly mention the required traits for conversion?

Reply:
Which traits do you mean? They use the same `FromPyObject` trait?

Reply:
Yes, and ToPyObject for returning back from functions.